runtime/src/deterministic.rs

```diff
@@ -853,18 +877,31 @@ impl crate::Spawner for Context {
         // Get metrics
         let (label, metric) = spawn_metrics!(self, future);
 
+        // Track parent-child relationship when supervision is requested
+        let parent_children = if self.spawn_config.is_supervised() {
+            Some(self.children.clone())
+        } else {
+            None
+        };
+
         // Set up the task
         let executor = self.executor();
+        self.spawn_config = SpawnConfig::default();
```

Comment:
An alternative approach that avoids this complexity (as discussed during standup) is to create some sort of intermediate spawner object (on which `spawn` is called):
```
context.spawner().dedicated().spawn_ref()
```

This would make it clear that these configs are never actually attached to the context object (or persist) anywhere. It would, however, complicate the flow of context into spawn (and is not the current preference).